server/backend/neighbors.py

```diff
@@ -172,39 +182,36 @@ def query_model_for_tok(
     result = []
     word_vectors = model if use_keyedvec else model.wv
 
-    cutoff_index = min(
-        map(
-            lambda x: (
-                999999
-                if word_vectors.get_vecattr(x[1], "count") > word_freq_count_cutoff
-                else x[0]
-            ),
-            enumerate(word_vectors.index_to_key),
-        )
-    )
-
     # Check to see if token is in the vocab
     vocab = set(word_vectors.key_to_index.keys())
 
     if tok in vocab:
         # If it is grab the neighbors
         # Gensim needs to be > 4.0 as they enabled neighbor clipping (remove words from entire vocab)
-        word_neighbors = word_vectors.most_similar(
-            tok,
-            topn=neighbors,
-            clip_end=cutoff_index,
-        )
+        word_neighbors = word_vectors.most_similar(tok, topn=neighbors)
 
         # Append neighbor to word_neighbor_map
         for neighbor in word_neighbors:
-            result.append(neighbor[0])
+            word_neighbor = neighbor[0]
+            
+            # Convert tags that contain the following pattern
+            # disease_mesh_####### or chemical_mesh_#######
+            if "mesh" in word_neighbor:
+                word_neighbor = "_".join(word_neighbor.split("_")[1:])
+                
+            # Insert tagged suffix to show users that
+            # some concpets are tagged and some concepts are missed
+            # example: mcf-7 is a cellline but the token itself appears as well 
+            if word_neighbor in concept_id_mapper_dict:
+                word_neighbor = f"{concept_id_mapper_dict[word_neighbor]}_(tagged)"
```

Comment:
So, how does this work downstream? I assume that these tokens are going to show up as "XXX_(tagged)" in the frontend, unless Vince strips out that suffix. Perhaps something a little more structured would work, like a tuple that consists of `(token, is_tagged)` or a dict like `{'token': 'XXX', 'is_tagged': False}`.

Also, just to clarify, this is so people see things like "homo sapiens_(tagged)" rather than "species_9606" in the results, right?

Reply:
>So, how does this work downstream? I assume that these tokens are going to show up as "XXX_(tagged)" in the frontend, unless Vince strips out that suffix. Perhaps something a little more structured would work, like a tuple that consists of (token, is_tagged) or a dict like {'token': 'XXX', 'is_tagged': False}.

You got it! If we do the latter than we need to let the user know that this token was tagged. Reason for this necessity is that some of the tokens were missed by the tagger but the represent the same thing (i.e.a549 shows up but its tagged id cvcl_0023 also shows up). I'm sure this is a simple fix for Vince if we want this on the front end.

>Also, just to clarify, this is so people see things like "homo sapiens_(tagged)" rather than "species_9606" in the results, right?

Bingo. That's exactly right